### PR TITLE
Clean up clippy

### PR DESCRIPTION
**Pull request by gjcolombo**
Today I learned that my Neovim rust-analyzer configuration has been busted for months without my consciously noticing. Celebrate fixing it (which gets clippy warnings to show up in my diagnostics pane) by clippy-cleaning the repo.

### CODE REVIEW

File: lib/propolis/src/hw/virtio/softnpu.rs [hunk 1]
```diff
@@ -110,6 +110,8 @@ pub struct SoftNpu {
 unsafe impl Send for SoftNpu {}
 unsafe impl Sync for SoftNpu {}
 
+type LoadedP4Program = (Library, Box<dyn Pipeline>);
```
Comment (gjcolombo) [inline lib/propolis/src/hw/virtio/softnpu.rs:113]: @rcgoodfellow - is this a reasonable name for this type given its contents/function? I added this to satisfy the `type_complexity` lint; let me know if you'd instead prefer for me remove this type alias and just suppress the complexity warning.

Comment (rcgoodfellow) [reply lib/propolis/src/hw/virtio/softnpu.rs:113]: This looks good to me.

**Review by rcgoodfellow — APPROVED** [review]
SoftNpu and P9fs bits LGTM.

**Comment by gjcolombo** [comment]
One of the changes to nvme/queue.rs is broken. Investigating.

**Review by pfmooney — APPROVED** [review]

**Pull request merged**